woocommerce/woocommerce-ios
# [Mobile Payments] Add Stripe account remote

Closes: #5530 

fyi @malinajirka 

@joshheald @jaclync - just one review is needed, but both are you are welcome - thank you :)

### Description
- Adds a remote and fetching of account information from the Stripe extension.
- Relies on https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2200 available in 5.9.0
- **HOWEVER** - Note that the endpoint is incomplete. If you were to modify PaymentGatewayAccountStore to fetch from the endpoint, you'd find the is_live and test_mode keys absent in the response from the endpoint.
- So... although this PR adds the remote, the remote is NOT used by the application (yet)
- When https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2262 is resolved, account fetching should Just Work TM
- Also adds full unit test coverage

### Nuances
- Don't let the change count scare you. Most of that is JSON for the unit test responses.
- StripeAccount looks a lot like WCPayAccount, and not unexpectedly. Both the Stripe extension and the WooCommerce Payments extension both rely on Stripe to do their work.
- Right now, this includes both structs exposing status as a shared enum, WCPayAccountStatusEnum. This avoids modifying PaymentGatewayAccount at this time. (Modifying PaymentGatewayAccount would touch a lot of code including persistence.) This isn't entirely inappropriate since both extensions have Stripe as their backend, but it would be more appropriate to have WCPayAccountStatusEnum and a new StripeAccountStatusEnum map through an abstraction (Adapter pattern) for PaymentGatewayAccount status. I recommend deferring that to a separate PR.

### Testing instructions
- Ensure new unit tests pass
- Optional: Since this remote isn't used yet, you can, as a quick test, add the following to PaymentGatewayAccountStore (note that this will at this time result in an error as is_live and test_mode keys are absent). You'll need to install the Stripe extension and enter your private and public test keys for a Stripe developer account in the extension settings.

```
--- a/Yosemite/Yosemite/Stores/PaymentGatewayAccountStore.swift
+++ b/Yosemite/Yosemite/Stores/PaymentGatewayAccountStore.swift
@@ -11,8 +11,11 @@ public final class PaymentGatewayAccountStore: Store {
     ///
     private let remote: WCPayRemote
 
+    private let stripeRemote: StripeRemote
+
     public override init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network) {
         self.remote = WCPayRemote(network: network)
+        self.stripeRemote = StripeRemote(network: network)
         super.init(dispatcher: dispatcher, storageManager: storageManager, network: network)
     }
 
@@ -72,6 +75,21 @@ private extension PaymentGatewayAccountStore {
                 return
             }
         }
+
+        stripeRemote.loadAccount(for: siteID) { [weak self] result in
+            guard let self = self else {
+                return
+            }
+
+            switch result {
+            case .success(let stripeAccount):
+                print("==== \(stripeAccount)")
+                return
+            case .failure(let error):
+                print("==== \(error)")
+                return
+            }
+        }
     }
```

### Screenshots
N/A - no user facing aspects


---
